spec/queries/follow_ups_query_spec.rb

```diff
@@ -0,0 +1,248 @@
+
+      Timecop.freeze("May 1st 2021") do
+        create(:blood_pressure, recorded_at: "February 10th 2021", facility: facility_1, patient: patient, user: user_1)
+        create(:blood_pressure, recorded_at: "March 5th 2021", facility: facility_1, patient: patient, user: user_1)
+        create(:blood_pressure, recorded_at: "March 5th 2021", facility: facility_1, patient: patient_2, user: user_1)
+        create(:blood_pressure, recorded_at: "March 20th 2021", facility: facility_1, patient: patient, user: user_2)
+        create(:blood_pressure, recorded_at: 1.month.ago, facility: facility_2, patient: patient)
+
+        query = described_class.new(facility_1, :month, group_by: "blood_pressures.user_id")
+        expected = {
+          Period.month("Feb 1st 2021") => {user_1.id => 1, user_2.id => 0},
+          Period.month("March 1st 2021") => {user_1.id => 2, user_2.id => 1}
+        }
+        expect(query.hypertension).to eq(expected)
+      end
+    end
+  end
+
+  context "follow ups" do
+    let(:reg_date) { Date.new(2018, 1, 1) }
+    let(:first_follow_up_date) { reg_date + 1.month }
+    let(:second_follow_up_date) { first_follow_up_date + 1.day }
+    let(:current_user) { create(:user) }
+    let(:current_facility) { create(:facility, facility_group: current_user.facility.facility_group) }
+    let(:follow_up_facility) { create(:facility, facility_group: current_user.facility.facility_group) }
+    let(:hypertensive_patient) { create(:patient, registration_facility: current_facility, recorded_at: reg_date) }
+    let(:diabetic_patient) { create(:patient, :diabetes, registration_facility: current_facility, recorded_at: reg_date) }
+
+    before do
+      2.times do
+        create(:blood_sugar,
+          :with_encounter,
+          facility: current_facility,
+          patient: diabetic_patient,
+          user: current_user,
+          recorded_at: first_follow_up_date)
+        create(:blood_pressure,
+          :with_encounter,
+          patient: hypertensive_patient,
+          facility: current_facility,
+          user: current_user,
+          recorded_at: first_follow_up_date)
+      end
+
+      # visit at a facility different from registration
+      create(:blood_pressure,
+        :with_encounter,
+        patient: hypertensive_patient,
+        facility: follow_up_facility,
+        user: current_user,
+        recorded_at: first_follow_up_date)
+
+      # diabetic patient following up with a BP
+      create(:blood_pressure,
+        :with_encounter,
+        patient: diabetic_patient,
+        facility: current_facility,
+        user: current_user,
+        recorded_at: first_follow_up_date)
+
+      # another follow up in the same month but another day
+      create(:blood_pressure,
+        :with_encounter,
+        patient: hypertensive_patient,
+        facility: current_facility,
+        user: current_user,
+        recorded_at: second_follow_up_date)
+    end
+
+    describe ".follow_ups" do
```

Comment:
Since this is testing `Patient.follow_ups_by_period`, shouldn't these be in `Patient` instead?
Alternatively, we could test a call to `FollowUpsQuery` here.